src/main.rs

```diff
@@ -50,7 +50,37 @@ ow=0:or=0;38;5;16;48;5;203:no=0:ex=1;38;5;203:cd=0;38;5;203;48;5;236:mi=0;38;5;1
 ";
 
 fn run() -> Result<ExitCode> {
-    let matches = app::build_app().get_matches_from(env::args_os());
+    let mut app = app::build_app();
+
+    // When encountering parsing errors, print error message and exit
+    let matches = match app.get_matches_from_safe_borrow(env::args_os()) {
+        Ok(matches) => matches,
+        Err(err) => {
+            eprintln!("{}", err);
+            return Ok(ExitCode::GeneralError)
+        }
+    };
+
+    // Print short and long help
+    if matches.is_present("help") {
+        let _ = app.print_help();
+        // HACK: It seems like `clap` does not add a newline after the help
+        //       output when using `print_help`. Therefore add a newline here.
+        //       Note, that the `after_help` string already adds a newline to generate
+        //       an empty line before the next prompt
```

Comment:
Thanks for the explanation, but I don't think we need to add it here. Especially not twice. And I don't consider it a "HACK". `print_help` doesn't print a newline, so we add one.

Reply:
From the [Clap documentation](https://docs.rs/clap/2.31.0/clap/struct.App.html#method.print_help):

> Prints the full help message to `io::stdout()` using a `BufWriter` using the same method as if someone ran `--help` to request the help message

So if I handle `-h | --help` manually and then call `print_help()`, I'd expect it to print the help message with a newline at the end, just. Because the documentation says that it would *use the same method as if someone ran `--help`* (which is obviously not the case), I'd consider this a bit as a hack to achieve what the `print_help` method has promised to do.

I can totally understand that you don't want to have a comment that says `HACK` in your tool. I use such comments to underline that this is a piece of code that works around some issue (mostly with external libraries) that might cause issues in the future, when the called functionality is fixed. So in this case, `clap` might decide to add the missing newline which would result in two empty lines in `fd`.

If you still want this comment to be removed, I'll of course do that. It is your tool after all :)

Reply:
> using the same method

Well, okay. Then it's a question of whether or not this is a bug in `clap`. According to [this](https://github.com/clap-rs/clap/issues/1536), it looks like it is. Apparently it will be fixed in v3, but that has not been released yet.

In that case, the additional newline is a workaround for some upstream bug. I always considered a "hack" to be something else: a really ugly piece of code that does the right thing in principle, but in a horrible way. I don' think that's the case here. But now we're just down to nomenclature :smile:. I would personally prefer "workaround", but I'm also okay with the current wording, if it's really a bug in clap.

Reply:
Workaround is reasonable, I've changed it 😉